lib/Parser/parse.cpp

```diff
@@ -7708,7 +7708,7 @@ void PidRefStack::TrackAssignment(charcount_t ichMin, charcount_t ichLim)
         {
             return;
         }
-        Assert(this->GetIchLim() >= ichLim);
+        Assert(ichMin <= this->GetIchMin() && this->GetIchLim() <= ichLim)
```

Comment:
Oh.  `#define AssertMsg` (in Assertions.h) should be wrapped in `do { ... } while (false)` to force a semicolon to come after the usage of the macro.